docs/reference/migration/migrate_8_0/security.asciidoc

```diff
@@ -112,6 +112,37 @@ Specifying this setting in a transport profile in `elasticsearch.yml` will
 result in an error on startup.
 ====
 
+[discrete]
+[[saml-realm-nameid-changes]]
+.The `nameid_format` SAML realm setting has no default value.
+[%collapsible]
+====
+*Details* +
+In SAML, Identity Providers (IdPs) can either be explicitly configured to
+release a `NameID` with a specific format, or configured to attempt to conform 
+with the requirements of a Service Provider (SP). The SP declares its
+requirements in the `NameIDPolicy` element of a SAML Authentication Request.
+In {es}, the `nameid_format` SAML realm setting controls the `NameIDPolicy`
+value.
+
+Previously, the default value for `nameid_format` was
+`urn:oasis:names:tc:SAML:2.0:nameid-format:transient`. This setting created
+authentication requests that required the IdP to release `NameID` with a
+`transient` format.
+
+The default value has been removed, which means that {es} will create SAML Authentication Requests by default that don't put this requirement on the
+IdP. If you want to retain the previous behavior, set `nameid_format` to
+`urn:oasis:names:tc:SAML:2.0:nameid-format:transient`.
+
+*Impact* +
+If you currently don't configure `nameid_format` explicitly, it's possible
+that your configuration won't match the `NameID` format that your IdP expects.
```

Comment:
Sorry for drive-by comment but should this be
```suggestion
that your configuration won't match the `NameID` format that your IdP releases.
```
?
My understanding is that if `nameid_format` is not explicitly configured, in a newer version of ES, the idP is free to release a `NameID` with a format by its own choice and this format may not work with the ES SAML configuration?

Reply:
@ywangd, I think you're correct in using "releases" here. Looking at the [SAML troubleshooting guide](https://www.elastic.co/guide/en/elasticsearch/reference/7.14/trb-security-saml.html), we indicate that:

> This is controlled by the nameid_format configuration parameter in elasticsearch.yml, which if not set defaults to urn:oasis:names:tc:SAML:2.0:nameid-format:transient. This instructs the Identity Provider to return a NameID with that specific format in the SAML Response. If the SAML Identity Provider cannot grant that request, for example because it is configured to release a NameID format with urn:oasis:names:tc:SAML:2.0:nameid-format:persistent format instead, it returns this error indicating an invalid NameID policy.

Reply:
> My understanding is that if nameid_format is not explicitly configured, in a newer version of ES, the idP is free to release a NameID with a format by its own choice and this format may not work with the ES SAML configuration?

`nameid_format` doesn't configure what should be matched in what the IDP releases though, so why does `release` make more sense here?

In any case ,we can only exactly match NameID with persistent format, we have the `nameid:persistent` magic string value that matches the NameID in the SAML response, but only if it is with persistent format. As such if you were already using 
```
atttributes.principal: nameid:persistent
```

then you either have already set `nameid_format: "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent"` in your config, or your IDP is configured to release a certain format no matter what. In both these cases , your setup will continue to work. 

I chose `expects` on purpose here to capture the possibility that the IDP is configured to release a NameID format based on what's in the NameID Policy (and it's own constraints), and the default change causes the SP to request a different format so the IDP cannot grant that request (because it violates it's own constraints) 

@lockewritesdocs I don't think there is something in this text that sways us one way or the other here, but it's a good reminder that this text is now stale and we need to update it for 8.0.0

Reply:
Thanks @jkakavas for the explanation. Apperantly my previous understanding on how idP responds to NameID request is wrong. Just one follow-up question on:

> I chose expects on purpose here to capture the possibility that the IDP is configured to release a NameID format based on what's in the NameID Policy (and it's own constraints), and the default change causes the SP to request a different format so the IDP cannot grant that request (because it violates it's own constraints)

ES's previous default for nameid is `transient`, now it is removed, which means a request from ES will not require certain nameid format. Are you saying that some idP could fail to respond when the SP does not explicitly specify a nameid format?

Reply:
I find this wording to be strange as well. 

> If you currently don't configure `nameid_format` explicitly, it's possible
> that your configuration won't match the `NameID` format that your IdP expects.

_"If you currently don't configure `nameid_format` explicitly"_ then the behaviour will change from setting a NameIDPolicy with a value of transient, to no longer setting a NameIDPolicy.

If the IdP is expecting a NameIDPolicy, then this will be a problem. 
But this _"your configuration won't match the `NameID` format that your IdP expects"_ doesn't read like that because it implies that the configuration has a "`NameID` format", when the problem (if there is one) will be that the configuration lacks a "`NameID` format".

If I understand your intent correctly, I think we want something like:
```suggestion
that your IdP will reject authentication request from {es} because the requests 
do not specify a `NameID` format (and your IdP is configured to expect one).
```

Reply:
I should rewrite 

> I chose expects on purpose here to capture the possibility that the IDP is configured to release a NameID format based on what's in the NameID Policy (and it's own constraints), and the default change causes the SP to request a different format so the IDP cannot grant that request (because it violates it's own constraints)

to 

> I chose expects on purpose here to capture the possibility that the IDP is configured to release a NameID format based on what's in the NameID Policy (and it's own constraints), and the default change causes the SP to not request a specific format so the IDP cannot grant that request because it has no fallback/default for what format to use.

I'll flick you a link from a slack thread where I had a contest with Tim about "what's the most contrived example this could fail in".

As said, this is a contrived example, I know of no IDP implementations that behave this way. Most IDPs 

1) Either require the admin to configure the format of the NameID it should use and do not care about what's in the policy of the request ( vast majority )
2) or allow a set of formats to be configured and limit that by what is in the policy of the request

Reply:
Yes Tim that's the intent and I like your wording, I will adopt.